src/lib.rs

```diff
@@ -129,6 +221,23 @@ where
 
         Ok(())
     }
+
+    /// Change configuration in register
+    ///
+    /// Helper function to update a particular part of a register without
+    /// affecting other parts of the register that might contain desired
+    /// configuration. This allows the `L3gd20` struct to be used like
+    /// a builder interface when configuring specific parameters.
+    fn change_config(&mut self, reg: Register, mask: u8, new_value: u8) -> Result<&mut Self, E> {
```

Comment:
I find a bit error prone the fact that the caller must take care of providing a properly shifted value *and* mask. Could this be refactorod into a function that takes the new value, the bitfield position (the number of bits to shift) and the width of the bitfield (from which the mask can be derived)

Reply:
Also to not add magic numbers to the `set_*` methods static methods like `Scale::pos()` and `Scale::width()` could be added to `Scale`, `ODR` and `Bandwidth` and used in the `set_*` methods.

Reply:
That is seems like a very sensible approach. Do you want it generalized to a trait? Something like:
```rust
trait BitValue {
    /// Bit shift offset
    fn shift(&self) -> u8;
    /// Bit width
    fn width(&self) -> u8;
    /// Bit value
    fn value(&self) -> u8;
}

impl BitValue for Odr {
    fn shift(&self) -> u8 { 6 }
    fn width(&self) -> u8 { 0x03 }
    fn value(&self) -> u8 { *self as u8 }
}

...
    fn change_config<T: BitValue>(&mut self, reg: Register, value: T) -> Result<&mut Self, E> {
        let mask = value.width() << value.shift();
        let bits    = (value.value() << value.shift()) & mask;
        // Then same as previous implementation
        let current = self.read_register(reg)?;
        let masked = current & !mask;
        let new_reg = masked | bits;
        self.write_register(reg, new_reg)?;
        Ok(self)
    }
```
Maybe change the above to dynamic dispatch to reduce binary size. Since this method will only be used during initialization it doesn't have to be the quickest.

Alternatively:
```rust
fn change_config(&mut self, reg: Register, shift: u8, width: u8, value: u8) -> Result<&mut Self, E> {
    // Same as above without the method calls to `width()`, `shift()` and `value()`
}
```